pkg/build/controller/image_change_controller_test.go

```diff
@@ -110,21 +111,18 @@ func TestNewImageID(t *testing.T) {
 	buildcfg := mockBuildConfig("registry.com/namespace/imagename", "registry.com/namespace/imagename", "testImageRepo", "testTag")
 	imagerepo := mockImageRepo("testImageRepo", "registry.com/namespace/imagename", map[string]string{"testTag": "newImageID123"})
 	controller := mockImageChangeController(buildcfg)
-	err := controller.HandleImageRepo(imagerepo)
-	buildCreator := controller.BuildCreator.(*mockBuildCreator)
+	buildConfigInstantiator := controller.BuildConfigInstantiator.(*mockBuildConfigInstantiator)
 	buildConfigUpdater := controller.BuildConfigUpdater.(*mockBuildConfigUpdater)
 
+	err := controller.HandleImageRepo(imagerepo)
 	if err != nil {
 		t.Errorf("Unexpected error %v from HandleImageRepo", err)
 	}
-	if buildCreator.build == nil {
-		t.Fatal("Expected new build when new image was created!")
-	}
-	if buildCreator.build.Parameters.Strategy.DockerStrategy.Image != "registry.com/namespace/imagename:newImageID123" {
```

Comment:
i think you've lost something in these tests by not explicitly checking the Image field.